apps/cms.c

```diff
@@ -897,8 +900,10 @@ int cms_main(int argc, char **argv)
                     goto end;
             }
 
-            res = EVP_PKEY_CTX_ctrl(pctx, -1, -1,
-                        EVP_PKEY_CTRL_CIPHER, EVP_CIPHER_nid(cipher), NULL);
+            ctx_params[0] = OSSL_PARAM_construct_utf8_string("cipher",
+                cipher_name, strlen(cipher_name));
+            ctx_params[1] = OSSL_PARAM_construct_end();
+            res = EVP_PKEY_CTX_set_params(pctx, ctx_params);
             if (res <= 0 && res != -2)
```

Comment:
You can query if its settable via..
EVP_PKEY_CTX_settable_params

Reply:
Fix applied.